Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BaseShapeComponentMode.cpp

```diff
@@ -0,0 +1,369 @@
+        {
+            return false;
+        }
+
+        if (mouseInteraction.m_mouseEvent == ViewportInteraction::MouseEvent::Wheel &&
+            mouseInteraction.m_mouseInteraction.m_keyboardModifiers.Ctrl())
+        {
+            const int direction = MouseWheelDelta(mouseInteraction) > 0.0f ? -1 : 1;
+            AZ::u32 currentModeIndex = static_cast<AZ::u32>(m_subMode);
+            AZ::u32 numSubModes = static_cast<AZ::u32>(ShapeComponentModeRequests::SubMode::NumModes);
+            AZ::u32 nextModeIndex = (currentModeIndex + numSubModes + direction) % m_subModes.size();
+            ShapeComponentModeRequests::SubMode nextMode = static_cast<ShapeComponentModeRequests::SubMode>(nextModeIndex);
+            SetShapeSubMode(nextMode);
+            return true;
+        }
+        return false;
+    }
+
+    AZStd::vector<ActionOverride> BaseShapeComponentMode::PopulateActionsImpl()
+    {
+        if (!m_allowAsymmetricalEditing)
+        {
+            return {};
+        }
+
+        ActionOverride setDimensionsModeAction;
+        setDimensionsModeAction.SetUri(SetDimensionsSubModeActionUri);
+        setDimensionsModeAction.SetKeySequence(QKeySequence(Qt::Key_1));
+        setDimensionsModeAction.SetTitle("Set Dimensions Mode");
+        setDimensionsModeAction.SetTip("Set dimensions mode");
+        setDimensionsModeAction.SetEntityComponentIdPair(GetEntityComponentIdPair());
+        setDimensionsModeAction.SetCallback(
+            [this]()
+            {
+                SetShapeSubMode(SubMode::Dimensions);
+            });
+
+        ActionOverride setTranslationOffsetModeAction;
+        setTranslationOffsetModeAction.SetUri(SetTranslationOffsetSubModeActionUri);
+        setTranslationOffsetModeAction.SetKeySequence(QKeySequence(Qt::Key_2));
+        setTranslationOffsetModeAction.SetTitle("Set Translation Offset Mode");
+        setTranslationOffsetModeAction.SetTip("Set translation offset mode");
+        setTranslationOffsetModeAction.SetEntityComponentIdPair(GetEntityComponentIdPair());
+        setTranslationOffsetModeAction.SetCallback(
+            [this]()
+            {
+                SetShapeSubMode(SubMode::TranslationOffset);
+            });
+
+        ActionOverride resetModeAction;
+        resetModeAction.SetUri(ResetSubModeActionUri);
+        resetModeAction.SetKeySequence(QKeySequence(Qt::Key_R));
+        resetModeAction.SetTitle("Reset Current Mode");
+        resetModeAction.SetTip("Reset current mode");
+        resetModeAction.SetEntityComponentIdPair(GetEntityComponentIdPair());
+        resetModeAction.SetCallback(
+            [this]()
+            {
+                ResetShapeSubMode();
+            });
+
+        return { setDimensionsModeAction, setTranslationOffsetModeAction, resetModeAction };
+    }
+
+    void BaseShapeComponentMode::RegisterActions(const char* shapeName)
+    {
+        AZStd::string category = AZStd::string::format("%s Component Mode", shapeName);
```

Comment:
`category` can be const.

Reply:
The intention of the following lines was to capitalize the first character of category, which would mean it couldn't be const. But I think what I had was actually broken anyway, and was just capitalizing another single character string which was a copy of the first character. I've fixed that, but it means that category can't be const.